### PR TITLE
error handling improved, context simplified

### PR DESCRIPTION
**Pull request by frairon**

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -285,6 +284,14 @@ func WithViewLogger(log logger.Logger) ViewOption {
 	}
 }
 
+// WithViewFullRecoverOnError sets the logger the view should use. By default, views
+// use the standard library logger.
+func WithViewFullRecoverOnError(log logger.Logger) ViewOption {
```
Comment (R053NR07) [inline options.go]: delete that.

File: partition_processor.go [hunk 2]
```diff
@@ -215,32 +259,51 @@ func (pp *PartitionProcessor) Stop() error {
...
 			case <-ctx.Done():
+				rerr = err
 				return
 			default:
 			}
 			panic(err)
 		}
 
-		asyncErrors = make(chan error, pp.opts.partitionChannelSize)
+		asyncErrors = make(chan error, 10)
```
Comment (R053NR07) [inline partition_processor.go]: Use channel size 1. (Other errors will still be logged)

File: partition_processor.go [hunk 3]
```diff
@@ -167,9 +187,30 @@ func (pp *PartitionProcessor) Setup(ctx context.Context) error {
...
+	// as the table is now recovered, we have to start handling stats requests
+	// separately during running
+	if pp.table != nil {
+		pp.runnerGroup.Go(func() error {
+			pp.table.handleStatsRequests(ctx)
+			return nil
+		})
+	}
+
+	for _, join := range pp.joins {
```
Comment (R053NR07) [inline partition_processor.go]: You may delete that :)

Comment (frairon) [reply partition_processor.go]: yep done.

**Pull request merged**